autoemulate/experimental/simulations/base.py

```diff
@@ -0,0 +1,164 @@
+from abc import ABC, abstractmethod
+
+import torch
+from tqdm import tqdm
+
+from autoemulate.experimental.data.utils import ValidationMixin
+from autoemulate.experimental.types import TensorLike
+from autoemulate.experimental_design import LatinHypercube
+
+
+class Simulator(ABC, ValidationMixin):
+    """
+    Base class for simulations. All simulators should inherit from this class.
+    This class provides the interface and common functionality for different
+    simulation implementations.
+    """
+
+    def __init__(
+        self,
+        parameters_range: dict[str, tuple[float, float]],
+        output_variables: list[str],
+    ):
+        """
+        Initialize the base simulator with parameter ranges
+          and optional output variables.
+
+        Parameters
+        ----------
+        parameters_range : dict
+            Dictionary mapping parameter names to their (min, max) ranges.
+        output_variables : list
+            Optional list of specific output variables to track.
+        """
+        # define param names and values
+        self._parameters_range = parameters_range
+        self._param_names = list(self._parameters_range.keys())
+        self._param_bounds = list(self._parameters_range.values())
+
+        # Output configuration
+        self._output_variables = (
+            output_variables if output_variables is not None else []
+        )
+        self._output_names: list[str] = []  # Will be populated after first simulation
```

Comment:
We should have an option for `_output_names` to be populated without having to run the simulation (I think this is quite specific to how the Naghavi simulator is implemented). This is the kind of information that could go in a `SimulatorMetadata` class if we have one.